src/test/java/ikysil/training/ws/api/v1/OrdersResourceTest.java

```diff
@@ -24,27 +30,170 @@
-    public void test() {
-        OrderDto request = new OrderDto();
-
+    public void savingOKTest() {
+        given().contentType(ContentType.JSON)
+                .body(order1, ObjectMapperType.JACKSON_2)
+                .when()
+                .post()
+                .then()
+                .statusCode(200);
         given().contentType(ContentType.JSON)
-                .body(request, ObjectMapperType.JACKSON_2)
+                .body(order2, ObjectMapperType.JACKSON_2)
                 .when()
                 .post()
                 .then()
-                .statusCode(INTERNAL_SERVER_ERROR.getStatusCode());
+                .statusCode(200);
+    }
+
+    @Test
+    public void saveOrderTest(){
+        OrderDto tempOrder = new OrderDto("666","thirty six", "nothing", "2660", "newuser66@gmail.com");
+        assertEquals(2, dao.findAll().size());
+        given().contentType(ContentType.JSON)
+                .body(tempOrder, ObjectMapperType.JACKSON_2)
+                .post();
+        assertEquals(3, dao.findAll().size());
+        assertNotNull(dao.findOrder(tempOrder.getId()));
+    }
+
+    @Test
+    public void getByIdTest() {
+        given().contentType(ContentType.JSON)
+                .when()
+                .get(order2.getId())
+                .then()
+                .body("userId", equalTo("4024"))
+                .body("deliveryInfo", equalTo(null))
+                .statusCode(200);
+        given().contentType(ContentType.JSON)
+                .when()
+                .get(order1.getId())
+                .then()
+                .body("userId", equalTo("3129"))
+                .body("deliveryInfo", equalTo(null))
+                .statusCode(200);
+    }
+
+
+    @Test
+    public void unvalidGetOrderTest() {
+        validationAPIMethodFactory("GET", "12345", 400);
+        validationAPIMethodFactory("GET", "qwe", 400);
+        validationAPIMethodFactory("GET", "abcd123456", 400);
+        validationAPIMethodFactory("GET", "as1234", 400);
+        validationAPIMethodFactory("GET", "agt1234", 400);
+    }
+
+    @Test
+    public void unvalidUpdateOrderTest() {
+        validationAPIMethodFactory("PUT", "12345", 400);
+        validationAPIMethodFactory("PUT", "qwe", 400);
+        validationAPIMethodFactory("PUT", "abcd123456", 400);
+        validationAPIMethodFactory("PUT", "as1234", 400);
+        validationAPIMethodFactory("PUT", "agt1234", 400);
+    }
+
+    @Test
+    public void unvalidDeleteOrderTest() {
+        validationAPIMethodFactory("DELETE", "12345", 400);
+        validationAPIMethodFactory("DELETE", "qwe", 400);
+        validationAPIMethodFactory("DELETE", "abcd123456", 400);
+        validationAPIMethodFactory("DELETE", "as1234", 400);
+        validationAPIMethodFactory("DELETE", "agt1234", 400);
+    }
+
+    //Method for reducing the code, I think this is a more flexible solution
+    private void validationAPIMethodFactory(String method, String orderId, int statusCode){
```

Comment:
why `*Factory`? Does it produce sth?
`testValidationError( )`?

Reply:
I have to work on the names i give